docs/trace/extending-the-sdk/README.md

```diff
@@ -362,6 +362,14 @@ register custom OpenTelemetry components into their `TracerProvider`s. These
 extension methods can target either the `TracerProviderBuilder` or the
 `IServiceCollection` classes. Both of these patterns are described below.
 
+**Note:** Libraries providing SDK plugins such as exporters, resource detectors,
+and/or samplers should take a dependency on the SDK (`OpenTelemetry.dll`)
+package. Library authors providing instrumentation should take a dependency on
+`OpenTelemetry.Api` or `OpenTelemetry.Api.ProviderBuilderExtensions` package.
+`OpenTelemetry.Api.ProviderBuilderExtensions` exposes interfaces for accessing
+the `IServiceCollection` which is a requirement for supporting the [.NET Options
+pattern](https://learn.microsoft.com/dotnet/core/extensions/options).
```

Comment:
hint that this is highly recommended, but not a requirement.

Reply:
Right below this there is more info about the options pattern and why authors should support it. Check that out and let me know if you still want more detail here.